### ISSUE TITLE
Calling resetForm() from Formsy onSubmit doesn't reset FormsyText

### ISSUE THREAD

**Issue by damonmaria** [mbrookes/formsy-material-ui#96]
formsy-material-ui@0.4.2, formsy-react@0.18.0, react@15.0.2, material-ui@0.15.0-beta.2

        <Formsy.Form
          onSubmit={(data, resetForm) => {
            addTask(data.task.trim())
            resetForm()
            // input.setState({ value: '' })
          }}
        >
          <FormsyText
            ref={(node) => {input = node}}
            name="task"
            floatingLabelText="Type to add a new task"
          />
        </Formsy.Form>
  
Calling resetForm() above does not reset the text input. However, the commented out line that uses input.setState({ value: '' }) does work.

**Comment by krizka** [comment]
Up. Same for me

**Comment by krizka** [comment]
Ok. I fixed this by extending FormsyText component with code: 
```
export default class FixedText extends FormsyText {
    componentWillReceiveProps(props) {
        this.setState({ value: this.getValue() || '' });
    }
}
```
Please fix it in FormsyText itself.

**Comment by mbrookes** [comment]
@krizka I'm kinda snowed, but feel free to submit a PR an dI'll look at it when I can. Any explanation on the rational behind the approach (step by step through the data flow) will help with a speedy review and merge.

**Comment by mbrookes** [comment]
Closed in #106.
